File: src/main/java/net/hectus/neobb/turn/default_game/mob/MobTurn.java
package net.hectus.neobb.turn.default_game.mob;

import net.hectus.neobb.player.NeoPlayer;
import net.hectus.neobb.turn.Turn;
import net.hectus.neobb.util.Utilities;
import org.bukkit.Material;
import org.bukkit.entity.LivingEntity;
import org.bukkit.inventory.ItemStack;

public abstract class MobTurn<T extends LivingEntity> extends Turn<T> {
    public MobTurn(NeoPlayer player) { super(null, null, player); }
    public MobTurn(T data, NeoPlayer player) { super(data, data.getLocation(), player); }

    @Override
    public ItemStack item() {
        return new ItemStack(Material.valueOf(Utilities.camelToSnake(Utilities.counterFilterName(getClass().getSimpleName())).toUpperCase() + "_SPAWN_EGG"));
    }
}

File: src/main/kotlin/net/hectus/neobb/turn/default_game/mob/MobTurn.kt
package net.hectus.neobb.turn.default_game.mob

import com.marcpg.libpg.storing.Cord
import net.hectus.neobb.player.NeoPlayer
import net.hectus.neobb.turn.Turn
import net.hectus.neobb.util.camelToSnake
import net.hectus.neobb.util.counterFilterName
import net.hectus.neobb.util.enumValueNoCase
import org.bukkit.Material
import org.bukkit.entity.LivingEntity
import org.bukkit.inventory.ItemStack

abstract class MobTurn<T : LivingEntity>(data: T?, cord: Cord?, player: NeoPlayer?) : Turn<T>(data, cord, player) {
    override fun item(): ItemStack = ItemStack(enumValueNoCase<Material>((this::class.simpleName ?: "unknown").counterFilterName().camelToSnake() + "_SPAWN_EGG"))
}
